data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/admin/Readiness.java

```diff
@@ -51,19 +69,50 @@ public Readiness(final DataProvider provider) {
       summary = "Get node readiness",
       description = "Returns 200 if the node is ready to accept traffic",
       tags = {TAG_TEKU},
+      queryParams = {
+        @OpenApiParam(name = TARGET_PEER_COUNT, description = TARGET_PEER_COUNT_DESCRIPTION),
+      },
       responses = {
         @OpenApiResponse(status = RES_OK, description = "Node is ready"),
+        @OpenApiResponse(
+            status = RES_BAD_REQUEST,
+            description = "Cannot parse " + TARGET_PEER_COUNT + " parameter passed in request"),
         @OpenApiResponse(
             status = RES_SERVICE_UNAVAILABLE,
             description = "Node not initialized or having issues")
       })
   @Override
   public void handle(final Context ctx) throws Exception {
     ctx.header(Header.CACHE_CONTROL, CACHE_NONE);
-    if (!chainDataProvider.isStoreAvailable() || syncProvider.isSyncing()) {
+    if (!canParseTargetPeerCountParameter(ctx)) {
+      ctx.status(SC_BAD_REQUEST);
+    } else if (!chainDataProvider.isStoreAvailable()
+        || syncProvider.isSyncing()
+        || !reachedTargetPeerCount(ctx)) {
       ctx.status(SC_SERVICE_UNAVAILABLE);
     } else {
       ctx.status(SC_OK);
     }
   }
+
+  private Boolean canParseTargetPeerCountParameter(final Context ctx) {
+    try {
+      final Map<String, List<String>> parameters = ctx.queryParamMap();
+      return parameters.isEmpty() // Backwards compatibility
+          || (parameters.containsKey(TARGET_PEER_COUNT)
+              && getParameterValueAsLong(parameters, TARGET_PEER_COUNT) > 0);
+    } catch (final IllegalArgumentException ex) {
+      LOG.error("Cannot parse {} parameter in Readiness", TARGET_PEER_COUNT, ex);
```

Comment:
I've pushed a change so that we avoid doing this parsing twice and include the error message in the response body instead of as an error log.